promql/engine.go

```diff
@@ -404,15 +414,22 @@ func (ng *Engine) exec(ctx context.Context, q *query) (Value, storage.Warnings,
 	q.cancel = cancel
 
 	defer func() {
-		// queryLogger might have been swapped or removed since the query
-		// start.
+		ng.queryLoggerLock.RLock()
 		if l := ng.queryLogger; l != nil {
-			f := []interface{}{"statement", q.Statement().String()}
+			f := []interface{}{"query", q.q}
+			if eq, ok := q.Statement().(*EvalStmt); ok {
+				f = append(f,
```

Comment:
Shouldn't we also indicate if it failed?

Reply:
This is update now afaik

Reply:
I'm not seeing where you added this.

Reply:
ooh now I get it. You want to check if the query fails. On it.

Reply:
I added it.